### PR TITLE
Adapting to support  protobuf 3

### PR DESCRIPTION
**Pull request by Novemser**
In protobuf3, some fields do not support `hasXXX` method, as a workaround in our case, we first get the field and then check whether it's value is null.

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tikv-client-lib-java/154)

### CODE REVIEW

**Comment by Novemser** [comment]
Clean clone and TPCH test verified, no problem. @ilovesoup PTAL

File: src/main/java/com/pingcap/tikv/region/TiRegion.java [hunk 1]
```diff
@@ -128,15 +128,15 @@ public boolean contains(ByteString key) {
   }
 
   public boolean isValid() {
-    return peer != null && meta.hasStartKey() && meta.hasEndKey();
+    return peer != null && meta != null && hasEndKey() && hasEndKey();
   }
 
   boolean hasStartKey() {
-    return meta.hasStartKey();
+    return meta.getStartKey() != null;
```
Comment (ilovesoup) [inline src/main/java/com/pingcap/tikv/region/TiRegion.java]: If it's not optional, doesn't protobuf stub generate a default?
So this test might be useless. Please confirm.

Comment (Novemser) [reply src/main/java/com/pingcap/tikv/region/TiRegion.java]: In protobuf3, all fields have default value of 0, so this test seems useless.

Comment (Novemser) [reply src/main/java/com/pingcap/tikv/region/TiRegion.java]: https://developers.google.com/protocol-buffers/docs/proto3#default

**Review by birdstorm — APPROVED** [review]
LGTM

**Pull request merged**